### ISSUE TITLE
pik gemsync broken

### ISSUE THREAD

**Issue by vertiginous** [vertiginous/pik#33]
I am trying to move all my gems from the JRuby RC to the latest version by using gemsync.  Is this the preferred way?
Anyway, when trying to gemsync from jruby1.4RC to jruby1.4 I get the following errors:

C:\Documents and Settings\dominic e sisneros>pik use jruby
Select which Ruby you want:
1. 140: jruby 1.4.0RC1 (ruby 1.8.7 patchlevel 174) (2009-09-30 80c263b) (Java HotSpot(TM) Client VM 1.5.0_11) [x86-
java]
2. 140: jruby 1.4.0 (ruby 1.8.7 patchlevel 174) (2009-11-02 69fbfa3) (Java HotSpot(TM) Client VM 1.5.0_11) [x86-jav
a]
?  1

C:\Documents and Settings\dominic e sisneros>pik gemsync jruby
Select which Ruby you want:
1. 140: jruby 1.4.0RC1 (ruby 1.8.7 patchlevel 174) (2009-09-30 80c263b) (Java HotSpot(TM) Client VM 1.5.0_11) [x86-
java]
2. 140: jruby 1.4.0 (ruby 1.8.7 patchlevel 174) (2009-11-02 69fbfa3) (Java HotSpot(TM) Client VM 1.5.0_11) [x86-jav
a]
?  2

There was an error.
 Error: undefined method `ruby_exe' for #Pik::GemSync:0x32b3d40

  in: pik/commands/gemsync_command.rb:37:in `gem_cache'
  in: pik/commands/gemsync_command.rb:17:in`install_gems'
  in: pik/commands/gemsync_command.rb:11:in `execute'
  in: pik_runner:35

**Comment by vertiginous** [comment]
I fixed this bug.  I also changed the semantics of the command to sync the current version with the version given, instead of the other way around.  This makes the code simpler, and doesn't require a batchfile.  In the older implmentation of gemsync, if you encountered an error you could get switched to the other version without being notified.  It was confusing.

```
>pik ls
091: IronRuby 0.9.1.0 on .NET 2.0.0.0
092: IronRuby 0.9.2.0 on .NET 2.0.0.0
131: jruby 1.3.1 (ruby 1.8.6p287) (2009-06-15 2fd6c3d) (Java HotSpot(TM) Client
VM 1.6.0_14) [x86-java]
140: jruby 1.4.0 (ruby 1.8.7 patchlevel 174) (2009-11-02 69fbfa3) (Java HotSpot(TM) Client VM 1.6.0_14) [x86-java] *
186: ruby 1.8.6 (2009-03-31 patchlevel 368) [i386-mingw32]
186: ruby 1.8.6 (2009-08-04 patchlevel 383) [i386-mingw32]
191: ruby 1.9.1p243 (2009-07-16 revision 24175) [i386-mingw32]

>pik gemsync 1.3
** Syncing: 140: jruby 1.4.0 (ruby 1.8.7 patchlevel 174) (2009-11-02 69fbfa3) (Java HotSpot(TM) Client VM 1.6.0_14) [x86-java]
   with: 131: jruby 1.3.1 (ruby 1.8.6p287) (2009-06-15 2fd6c3d) (Java HotSpot(TM) Client VM 1.6.0_14) [x86-java]

** Installing rubyforge-2.0.3
Successfully installed json_pure-1.1.9
Successfully installed rubyforge-2.0.3
2 gems installed
** Gem rspec-1.2.6 already installed
** Gem rake-0.8.7 already installed
** Gem json_pure-1.1.9 already installed
** Gem jruby-openssl-0.5.2 already installed
** Installing hoe-2.3.3
Successfully installed hoe-2.3.3
1 gem installed
** Installing fastercsv-1.5.0
Successfully installed fastercsv-1.5.0
1 gem installed
```

I still need to write some tests for this command.

**Comment by vertiginous** [comment]
Fixed in  bfed28f409abdf726cc0d1964065a2da198c0c2c
